### PR TITLE
Disabled versioning button when site replication is enabled

### PR DESCRIPTION
**Pull request by bexsoft**
## What does this do?

Disabled versioning button when site replication is enabled

## How does it look?

Screen Shot 2022-05-03 at 23 19 44

Signed-off-by: Benjamin Perez <benjamin@bexsoft.net>

### CODE REVIEW

File: portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket.tsx [hunk 1]
```diff
@@ -355,7 +368,11 @@ const AddBucket = ({
...
                   label={"Versioning"}
-                  disabled={!distributedSetup || lockingEnabled}
+                  disabled={
+                    !distributedSetup ||
+                    lockingEnabled ||
+                    siteReplicationInfo.enabled
+                  }
                 />
               </Grid>
               <Grid item xs={12}>
```
Comment (prakashsvmx) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket.tsx:378]: there is a minor ui only issue: (With site replication enabled)
Enable Object Lock -> disable object Lock. (observe that the versioning toggle changes to on state).

suggestion for the fix
```javascript
<Grid item xs={12}>
                <FormSwitchWrapper
                  value="locking"
                  id="locking"
                  name="locking"
                  disabled={lockingFieldDisabled || !distributedSetup}
                  checked={lockingEnabled}
                  onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
                    enableObjectLocking(event.target.checked);
                    if (event.target.checked) {
                      addBucketVersioned(true);
                    }
                    if (siteReplicationInfo.enabled) {
                      addBucketVersioned(false);
                    }
                  }}
                  label={"Object Locking"}
                />
              </Grid>
```

**Comment by prakashsvmx** [comment]
@bexsoft Thank you for this PR.

File: portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket.tsx (hunk 1 above)
Comment (bexsoft) [reply portal-ui/src/screens/Console/Buckets/ListBuckets/AddBucket.tsx:395]: Added validation to code

**Comment by bexsoft** [comment]
Updated message to be less impresive 

Screen Shot 2022-05-04 at 14 05 16

**Review by dvaldivia — APPROVED** [review]

**Review by adfost — APPROVED** [review]

**Pull request merged**